paperqa/agents/search.py

```diff
@@ -326,6 +344,14 @@ async def _add_document() -> None:
             )
             raise
 
+    async def commit(self) -> None:
+        """Commit all pending changes to the index."""
+        if self._writer:
+            self._writer.commit()
+            self._writer.wait_merging_threads()
+            self._searcher = None
```

Comment:
Can you add a comment why we are wiping the searcher?

Reply:
No need to wipe the searcher actually